manim/scene/scene_file_writer.py

```diff
@@ -392,17 +393,14 @@ def finish(self):
                 self.flush_cache_directory()
             else:
                 self.clean_cache()
-        if file_writer_config["save_last_frame"]:
-            self.scene.update_frame(ignore_skipping=True)
-            self.save_final_image(self.scene.camera.get_image())
 
     def open_movie_pipe(self):
         """
         Used internally by Manim to initalise
         FFMPEG and begin writing to FFMPEG's input
         buffer.
         """
-        file_path = self.partial_movie_files[self.scene.num_plays]
+        file_path = self.partial_movie_files[self.scene.renderer.num_plays]
```

Comment:
The goal is to decouple the renderer from the scene, so it will have to be held here to allow for that. And besides that, `num_plays` is only used functionally in scene skipping, which isn't part of the Scene's goal of managing Mobjects/Animations.